public/app/plugins/datasource/mysql/sqlUtil.test.ts

```diff
@@ -0,0 +1,17 @@
+import { isValidIdentifier } from './sqlUtil';
+
+describe('isValidIdentifier', () => {
+  test.each([
+    { value: 'and', expected: false }, // Reserved keyword
+    { value: '1name', expected: false }, // Starts with value
+    { value: 'my-sql', expected: false }, // Contains not permitted character
+    { value: '$id', expected: false }, // $ sign shouldn't be the first character
+    { value: 'my sql', expected: false }, // Whitespace is not permitted
+    { value: 'mysql ', expected: false }, // Whitespace is not permitted at the end
```

Comment:
Any need to test whitespace prepending the string here?

Reply:
yeah that can be added.